app/nostra.rb

```diff
@@ -5,4 +5,47 @@
 require_relative './nostra/account.rb'
 
 module Nostra
+  def self.import_csv(csv_file, type = :one_money)
+    case type
+    when :one_money
+      add_by_csv(csv_file)
+    else
+      raise "#{type} csv not implemented for import!"
+    end
+  end
+
+  private
+
+  def self.add_by_csv(csv_file)
+    transaction_rows = CSVDataParser.transactions(csv_file)
+    add_transactions_by_account(transaction_rows)
+  end
+
+  def self.add_transactions_by_account(transaction_rows)
+    transactions_by_account = sort_by_from_account(transaction_rows)
+
+    transactions_by_account.each do |account|
+      new_account = Nostra::Account.new(account[:name])
+      new_account.add_transactions(account[:transactions])
+    end
+  end
+
+  def self.sort_by_from_account(transaction_rows)
+    sorted_transactions = transaction_rows.sort_by { |transaction| transaction[:from_account] }
+
+    collection = []
+    account_names = (sorted_transactions.each.collect { |transaction| transaction[:from_account] }).uniq
```

Comment:
is the `each` in this chain necessary? ... actually couldn't you just do `.uniq` on `sorted_transactions` directly?  I might be missing something here, but thats what it looks like at first glance... oh, i guess `[:from_account]` is getting the name out huh

Reply:
Yeah, it may be that I don't need the each. I'll play with it some more.